src/xrScriptEngine/script_profiler.cpp

```diff
@@ -0,0 +1,248 @@
+#include "pch.hpp"
+#include "script_profiler.hpp"
+
+CScriptProfiler::CScriptProfiler()
+{
+    m_active = false;
+    m_profile_level = 1;
+
+    // todo: Configuration of profile levels.
+    // todo: Configuration of profile levels.
+    // todo: Configuration of profile levels.
+
+    if (strstr(Core.Params, "-lua_profiler"))
+    {
+        start();
+    }
+}
+
+CScriptProfiler::~CScriptProfiler()
+{
+}
+
+void CScriptProfiler::start()
+{
+    if (m_active)
+    {
+        Msg("Tried to start active profiler, operation ignored");
+        return;
+    }
+
+    Msg("Starting lua scripts profiler");
+
+    m_active = true;
+
+    // todo: Reset?
+    // todo: Reset?
+    // todo: Reset?
+
+    // todo: Check JIT and warn? Allow turn it off with parameter?
+    // todo: Check JIT and warn? Allow turn it off with parameter?
+    // todo: Check JIT and warn? Allow turn it off with parameter?
+
+    // todo: Attach hook?
+    // todo: Attach hook?
+    // todo: Attach hook?
+}
+
+void CScriptProfiler::stop()
+{
+    if (!m_active)
+    {
+        Msg("Tried to stop inactive profiler, operation ignored");
+        return;
+    }
+
+    Msg("Stopping lua scripts profiler");
+
+    // todo: Reset?
+    // todo: Reset?
+    // todo: Reset?
+
+    // todo: Detach hook?
+    // todo: Detach hook?
+    // todo: Detach hook?
+}
+
+void CScriptProfiler::reset() 
+{ 
+    Msg("Reset profiler");
+
+    // todo;
+    // todo;
+    // todo;
+}
+
+void CScriptProfiler::log()
+{
+
+    u64 total_count = 0;
+    u64 total_duration = 0;
+
+    std::vector<decltype(m_profiling_portions)::iterator> entries;
```

Comment:
Always use xr_vector.
If the data isn't too big, you can even use `buffer_vector` with `xr_alloca` (there are examples in the code base).
```suggestion
    xr_vector<decltype(m_profiling_portions)::iterator> entries;
```

Reply:
For me it was 3-4k records in average. I believe it will be <10k pointers almost always